test/e2e/tests/test_rule.py

```diff
@@ -0,0 +1,238 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the EventBridge Rule resource
+"""
+
+import logging
+import time
+from typing import Dict
+
+import pytest
+
+from acktest import tags
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_eventbridge_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from e2e.tests.helper import EventBridgeValidator
+
+RESOURCE_PLURAL = "rules"
+
+CREATE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@pytest.fixture(scope="module")
+def event_bus():
+    resource_name = random_suffix_name("eventbridge-bus", 24)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["BUS_NAME"] = resource_name
+
+    resource_data = load_eventbridge_resource(
+        "eventbus",
+        additional_replacements=replacements,
+    )
+    logging.debug(resource_data)
+
+    # Create the k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, "eventbuses",
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    # Get latest event_bus CR
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield (ref, cr)
+
+    # Try to delete, if doesn't already exist
+    try:
+        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+        assert deleted
+    except:
+        pass
+
+@pytest.fixture(scope="module")
+def simple_rule(event_bus):
+    resource_name = random_suffix_name("eventbridge-rule", 24)
+    _, eb_cr = event_bus
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["BUS_NAME"] = eb_cr["spec"]["name"]
+    replacements["RULE_NAME"] = resource_name
+    replacements["EVENT_PATTERN"] = "{\\\"detail-type\\\":[\\\"ack-event\\\"]}"
```

Comment:
nit: Could make this easier using `json.dumps`:
```suggestion
    replacements["EVENT_PATTERN"] = json.dumps({
        "detail-type": ["ack-event"]
    })
```

Reply:
Oh god, thank you!

Reply:
Didn't work since, this is directory going to the resource yaml file..

Reply:
You'll need to wrap the YAML file variable with double quotes, so that it doesn't try to unwrap it as an object:
```yaml
EVENT_PATTERN: "$EVENT_PATTERN"
```

Reply:
I did:
```
apiVersion: eventbridge.services.k8s.aws/v1alpha1
kind: Rule
metadata:
  name: $RULE_NAME
spec:
  name: $RULE_NAME
  eventBusName: $BUS_NAME
  eventPattern: "$EVENT_PATTERN"
  tags:
  - key: env
    value: testing
```

Reply:
Let me give a shot and see if it will work with single quote, i can submit a new PR or fix in the next one

Reply:
Oh yeah I'm seeing this in the EKS controller advanced configuration PR as well. We just need some combination of single quotes and double quotes that don't cause string escape issues 👍🏼